doctr/models/factory/hub.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2021-2022, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+# Inspired by: https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/hub.py
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Any, Dict
+
+from huggingface_hub import HfApi, HfFolder, Repository
+
+from doctr.file_utils import is_tf_available, is_torch_available
+
+__all__ = ['login_to_hub', 'push_to_hf_hub']
+
+
+def login_to_hub() -> None:
+    """Login to huggingface hub
+    """
+
+    access_token = HfFolder.get_token()
+    if access_token is not None and HfApi()._is_valid_token(access_token):
+        print("Huggingface Hub token found and valid")
+        HfApi().set_access_token(access_token)
+    else:
+        subprocess.call(['huggingface-cli', 'login'])
+        HfApi().set_access_token(HfFolder().get_token())
+    # check if git lfs is installed
+    try:
+        subprocess.call(['git', 'lfs', 'version'])
+    except FileNotFoundError:
+        raise OSError('Looks like you do not have git-lfs installed, please install. \
+                      You can install from https://git-lfs.github.com/. \
+                      Then run `git lfs install` (you only have to do this once).')
+
+
+def _save_model_and_config_for_hf_hub(model: Any, save_dir: str, model_config: Dict[str, str]) -> None:
```

Comment:
Model config typing in incorrect (values can be non string)

Reply:
👍